vizro-core/docs/pages/user-guides/custom-figures.md

```diff
@@ -1,70 +1,235 @@
+            controls=[vm.Filter(column="names", selector=vm.RadioItems(title="Select a name"))],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+        Vizro().build(dashboard).run()
+        ```
+
+        1. Here we decorate our custom figure function with the `@capture("figure")` decorator.
+        2. The custom figure function needs to have a `data_frame` argument and return a `Dash` component.
+        3. We return a `html.H2` component that dynamically updates based on the selected name from the filter.
+        4. Our custom figure function `custom_kpi_card` now needs to be passed on to the `vm.Figure`.
+
+    === "app.yaml"
+        ```yaml
+        # Custom figures are currently only possible via python configuration
+        ```
+    === "Result"
+        [![CustomHTML]][CustomHTML]
+
+    [CustomHTML]: ../../assets/user_guides/figure/custom_html.png
+
+<!-- vale on -->
+
+
+#### Dynamic number of cards
+The example below shows how to create multiple cards created from a `pandas.DataFrame` where the
 number of cards displayed dynamically adjusts based on a `vm.Parameter`.
 
+<!-- vale off -->
 !!! example "Dynamic number of cards"
     === "app.py"
         ```py
-        from typing import List, Optional
+        from typing import Optional
 
+        import dash_bootstrap_components as dbc
         import pandas as pd
         import vizro.models as vm
+        from dash import dcc, html
         from vizro import Vizro
         from vizro.models.types import capture
 
         df = pd.DataFrame(
             {
                 "text": [
-                    "Lorem ipsum dolor sit amet, consetetur sadipscing no sea elitr sed diam nonumy.",
-                    "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
-                    "Sed diam voluptua. At vero eos et accusam et justo no duo dolores et ea rebum.",
-                    "Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.",
-                    "Lorem ipsum dolor sit amet, consetetur sadipscing no sea est elitr dolor sit amet.",
-                    "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
-                ]
-                * 2
+                            "Lorem ipsum dolor sit amet, consetetur sadipscing no sea elitr sed diam nonumy.",
+                            "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
+                            "Sed diam voluptua. At vero eos et accusam et justo no duo dolores et ea rebum.",
+                            "Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.",
+                            "Lorem ipsum dolor sit amet, consetetur sadipscing no sea est elitr dolor sit amet.",
+                            "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
+                        ]
+                        * 2
```

Comment:
This is super ugly, even if it's gone through black. 😱 I would un-black it or rewrite the dataframe somehow to make it easier to see, like just break it down:
```
text = [
   "blah", 
   "blah"
]
df = pd.DataFrame({"text": text * 2})
```